crates/sui-graphql-rpc/src/types/transaction_block_effects.rs

```diff
@@ -135,18 +154,24 @@ impl TransactionBlockEffects {
 
     /// The checkpoint this transaction was finalized in.
     async fn checkpoint(&self, ctx: &Context<'_>) -> Result<Option<Checkpoint>> {
-        let checkpoint = self.stored.checkpoint_sequence_number as u64;
-        ctx.data_unchecked::<PgManager>()
-            .fetch_checkpoint(None, Some(checkpoint))
-            .await
-            .extend()
+        if let Some(stored_tx) = &self.stored {
+            Ok(ctx
+                .data_unchecked::<PgManager>()
+                .fetch_checkpoint(None, Some(stored_tx.checkpoint_sequence_number as u64))
+                .await
+                .extend()?)
+        } else {
+            Ok(None)
+        }
     }
 
     // TODO: event_connection: EventConnection
 
     /// Base64 encoded bcs serialization of the on-chain transaction effects.
-    async fn bcs(&self) -> Option<Base64> {
-        Some(Base64::from(&self.stored.raw_effects))
+    async fn bcs(&self) -> Result<Option<Base64>, Error> {
+        let bytes = bcs::to_bytes(&self.native)
+            .map_err(|e| Error::Internal(format!("Error serializiing transaction effects: {e}")))?;
+        Ok(Some(Base64::from(&bytes)))
```

Comment:
- nit: typo in error
- If we have the stored effects, we should return the BCS bytes from there directly to avoid recomputing them unnecessarily.
- Return a `Result<...>` instead of a `Result<..., ...>` and `extend()` the error.
- It seems like this one we can make non-optional, because we'll always have it.

```suggestion
    async fn bcs(&self) -> Result<Base64> {
        let bytes = if let Some(stored) = &self.stored {
            self.stored.raw_effects.clone()
        } else {
            bcs::to_bytes(&self.native)
                .map_err(|e| Error::Internal(format!("Error serializing transaction effects: {e}")))
                .extend()?
        }
        
        Ok(Base64::from(bytes)))
```